terminus-ui/selection-list/src/selection-list-panel/selection-list-trigger.directive.ts

```diff
@@ -0,0 +1,798 @@
+    }
+
+    // If there are any subscribers before `ngAfterViewInit`, the selection list will be undefined.
+    // Return a stream that we'll replace with the real one once everything is in place.
+    return this.ngZone.onStable
+      .asObservable()
+      .pipe(take(1), switchMap(() => this.optionSelections));
+  });
+
+  /**
+   * Store whether the overlay is currently attached
+   */
+  private overlayAttached = false;
+
+  /**
+   * Store a reference to the overlay
+   */
+  public overlayRef: OverlayRef | null | undefined;
+
+  /**
+   * Old value of the native input
+   *
+   * NOTE: Used to work around issues with the `input` event on IE
+   */
+  private previousValue: string | number | null | undefined;
+
+  /**
+   * Store a reference to the portal
+   */
+  private portal: TemplatePortal | undefined;
+
+  /**
+   * Strategy that is used to position the panel
+   */
+  private positionStrategy!: FlexibleConnectedPositionStrategy;
+
+  /**
+   * The defined scroll strategy
+   */
+  private readonly scrollStrategy: () => ScrollStrategy;
+
+  /**
+   * Subscription to viewport size changes
+   */
+  private viewportSubscription = Subscription.EMPTY;
+
+  /**
+   * Define the default component ID
+   */
+  public readonly uid = `ts-selection-list-trigger-${nextUniqueId++}`;
+
+  /**
+   * The currently active option, coerced to TsOptionComponent type
+   */
+  public get activeOption(): TsOptionComponent | null {
+    if (this.selectionListPanel && this.selectionListPanel.keyManager) {
+      return this.selectionListPanel.keyManager.activeItem;
+    }
+
+    return null;
+  }
+
+  /**
+   * A stream of actions that should close the panel, including when an option is selected, on blur, and when TAB is pressed.
+   */
+  public get panelClosingActions(): Observable<TsOptionSelectionChange | null> {
+    return merge(
+      this.optionSelections,
+      this.selectionListPanel.keyManager.tabOut.pipe(filter(() => this.overlayAttached)),
+      this.closeKeyEventStream,
+      this.overlayRef.backdropClick(),
+    ).pipe(
+      // Normalize the output so we return a consistent type.
+      map(event => (event instanceof TsOptionSelectionChange ? event : null)),
+    );
+  }
+
+  /**
+   * Whether or not the panel is open
+   */
+  public get panelOpen(): boolean {
+    return this.overlayAttached && this.selectionListPanel.showPanel;
+  }
+
+  /**
+   * Reflect the settings from the parent
+   */
+  @Input()
+  public allowMultiple = false;
+
+  /**
+   * The `autocomplete` attribute to be set on the input element.
+   */
+  // NOTE: Input has specific naming since it is accepting a standard HTML data attribute.
+  // tslint:disable: no-input-rename
+  @Input('autocomplete')
+  public autocompleteAttribute = 'off';
+
+  /**
+   * Whether the trigger is disabled. When disabled, the element will act as a regular input and the user won't be able to open the panel.
+   */
```

Comment:
Based on previous form inputs, an autocomplete may or may not have possible options. Wasn't a direct feature requirement for selection-list but since it already existed I left it in.